astropy/io/fits/tests/test_hdulist.py

```diff
@@ -796,16 +796,16 @@ def test(mmap_a, mmap_b):
             hdu_b = fits.PrimaryHDU(data=arr_b)
             hdu_b.writeto(self.temp('test_b.fits'), overwrite=True)
 
-            hdul_a = fits.open(self.temp('test_a.fits'), mode='update',
-                               memmap=mmap_a)
-            hdul_b = fits.open(self.temp('test_b.fits'), memmap=mmap_b)
-            hdul_a[0].data = hdul_b[0].data
-            hdul_a.close()
-            hdul_b.close()
+            with fits.open(self.temp('test_a.fits'), mode='update',
+                           memmap=mmap_a) as hdul_a:
+                with fits.open(self.temp('test_b.fits'),
+                               memmap=mmap_b) as hdul_b:
+                    hdul_a[0].data = hdul_b[0].data
 
-            hdul_a = fits.open(self.temp('test_a.fits'))
+            with fits.open(self.temp('test_a.fits')) as hdul_a:
+                status = np.all(hdul_a[0].data == arr_b)
```

Comment:
One can just put the `assert` here - `with` will close the file even if the test fails.

Reply:
Ah, is that so? I wasn't sure if `__exit__` would run if `AssertionError` is raised first. OK, let me apply your suggestion and see if the Appveyor CI still passes. Thank you.

Reply:
I think `with` is essentially syntactic sugar for a `try/finally`, but definitely the point is that you don't have to worry about leaving things open, etc.!